Add support for Password Hardening

davidpil2002 opened the pull request:
<!--
    Please make sure you've read and understood our contributing guidelines:
    https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

    ** Make sure all your commits include a signature generated with `git commit -s` **

    If this is a bug fix, make sure your description includes "closes #xxxx",
    "fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
    issue when the PR is merged.

    If you are adding/modifying/removing any command or utility script, please also
    make sure to add/modify/remove any unit tests from the tests
    directory as appropriate.

    If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
    subcommand, or you are adding a new subcommand, please make sure you also
    update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
    your changes.

    Please provide the following information:
-->

#### What I did
Add Password Hardening CLI
#### How I did it
created the CLI by using YANG model generator, the YANG model can be found in the password hardening HLD https://github.com/Azure/SONiC/blob/master/doc/passw_hardening/hld_password_hardening.md#TestPlan
and also in sonic-buildimage will be merged in the path: src/sonic-yang-models/yang-models/sonic-passwh.yang
#### How to verify it
Manually:
you can use configurations command like"config passw-hardening policies <policy> <value>"  or 
"show passw-hardening policies" (more examples in the HLD.)
Auto:
1.There are unitest of each policy including good & bad flow in this commit, that should pass.

2.There are tests in sonic-mgmt repo in the path: sonic-mgmt/tests/passw_hardening/
the test are end to end test and the are testing the config/show CLI commands as well.
#### Previous command output (if the output of a command-line utility has changed)
(new feature)
#### New command output (if the output of a command-line utility has changed)

```
root@r-panther-13:/home/admin# show passw-hardening policies
STATE    EXPIRATION    EXPIRATION WARNING    HISTORY    LEN MAX    LEN MIN    USERNAME PASSW MATCH    LOWER CLASS    UPPER CLASS    DIGITS CLASS    SPECIAL CLASS
-------  ------------  --------------------  ---------  ---------  ---------  ----------------------  -------------  -------------  --------------  ---------------
enabled      30           10                   4         100        30               false                 true            true            true          true

```

dgsudharsan commented on show/plugins/sonic-passwh_yang.py:
```diff
@@ -0,0 +1,155 @@
+
+import click
+import tabulate
+import natsort
```
Please remove natsort as it throws unused import.

dgsudharsan commented on tests/passw_hardening_test.py:
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
```
Why do we need tests for auto-generated CLI? Is it because of the coverage?

davidpil2002 replied:
yes

liat-grozovik commented:
@liuh-80, @qiluo-msft kindly reminder to review and signoff

qiluo-msft commented:
The latest coverage requirement (DIFF_COVER_CHECK_THRESHOLD) is 80%. Could you improve the coverage?

zhangyanzhao commented:
Liat will check with Qi on the review.

davidpil2002 commented on config/plugins/sonic-passwh_yang.py:
```diff
@@ -0,0 +1,537 @@
+"""
+Autogenerated config CLI plugin.
```
Should be an HLD/DOC about how to use the tool, I removed the augenerated comment

davidpil2002 commented:
> The latest coverage requirement (DIFF_COVER_CHECK_THRESHOLD) is 80%. Could you improve the coverage?

I added more unitest and removed unused code.
the cov is now more than the threshold

davidpil2002 commented:
answered all the comment, the code is ready to be review and merge
@qiluo-msft 
@liat-grozovik

qiluo-msft commented:
@liuh-80 Could you review this PR?

davidpil2002 commented:
@liat-grozovik pls can you help or derived me about this EasyCLA authorization issue?

liat-grozovik commented:
/easycla

liat-grozovik commented:
@liuh-80 we need to close this feature review as it is for 202205. can you please help to approve the PR?

liuh-80 commented:
/easycla

liuh-80 commented:
@liat-grozovik, please fix the easycla issue, then I can approve this PR.

liuh-80 reviewed (approved):
I will approve again after fix easycla issue.

davidpil2002 commented:
/easycla

liuh-80 commented:
> * ❌ The commit ([6b50c67](https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

Please fix user name and emaill address issue on this commit:

❌ The commit (https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

zhangyanzhao commented:
@davidpil2002 can you please sign the EasyCLA to unblock the merge process? Thanks. @liat-grozovik  for viz

davidpil2002 commented:
/easycla

davidpil2002 commented:
/easycla

davidpil2002 commented:
/easycla

davidpil2002 commented:
/easycla

liuh-80 commented:
@davidpil2002 , the CLA issue caused by this commit: https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f

In this commit, your ID is 'davidpil'

In other commit which does not has the CLA issue, your ID is 'davidpil2002', so please check and fix your ID in that commit, then CLA will pass.

davidpil2002 commented:
> @davidpil2002 , the CLA issue caused by this commit: [6b50c67](https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f)
> 
> In this commit, your ID is 'davidpil'
> 
> In other commit which does not has the CLA issue, your ID is 'davidpil2002', so please check and fix your ID in that commit, then CLA will pass.

I fixed this issue by creating a new PR with similar name:https://github.com/sonic-net/sonic-utilities/pull/2338
